java/flight/flight-grpc/src/main/java/org/apache/arrow/flight/FlightGrpcUtils.java

```diff
@@ -49,10 +143,10 @@ public static BindableService createFlightService(BufferAllocator allocator, Fli
   /**
    * Creates a Flight client.
    * @param incomingAllocator  Memory allocator
-   * @param channel provides a connection to a gRPC server
+   * @param channel provides a connection to a gRPC server. Will not be closed on closure of the returned FlightClient.
```

Comment:
This would be a breaking change - can we do this in a separate method?

Reply:
To clarify, introduce a new method for creating flight clients which uses the proxy? To me that persists the issue that this was intended to solve, which is that closing a client closes the channel that was passed to it when that channel may be used to create multiple clients. Otherwise, I would suggest this simply be doc-fixed by noting that the client takes ownership and will close the channel upon closure.

Reply:
Yes - how does that not solve the problem? The caller can choose whether the FlightClient takes ownership of the channel or not by calling different methods. But if the current method is modified, then any existing code will suddenly silently leak channels.